config/manifests/bases/azure-service-operator.clusterserviceversion.yaml

```diff
@@ -165,6 +165,14 @@ spec:
       kind: KeyVault
       name: keyvaults.azure.microsoft.com
       version: v1alpha1
+    - description: Add an AAD user to an existing MySQL database.
```

Comment:
Just confirming -- this file is generated somehow is that right?

Reply:
No, unfortunately it isn't. If the CRDs aren't added here they'll still show up in the output cluster service version, but without any description. I don't think there's a downside to that other that in the Openshift UI for creating resources there won't be any explanatory text, but I figured I'd add it here.

There are doc comments on the Go types for the resources, but controller-gen puts that into the schema, and the `operator-sdk bundle` command doesn't use them to populate this field. (The existing ones on our resources aren't great, but we could update them if they were used.) As far as I can tell there isn't a description field on the [CRD k8s type](https://kubernetes.io/docs/reference/kubernetes-api/extend-resources/custom-resource-definition-v1/) for describing the resource as a whole.

A possible option would be to add another build step before this to extract the schema descriptions from the files in  `config/crd/bases`, and compile them into a patch that got stitched into the CSV file that then gets fed into `operator-sdk bundle`? (Plus updating the existing doc comments.)

Reply:
Yeah, might be worth filing an issue for this. What I'm seeing that is awkward here is:
1. If I add a new resource, how do I know I need to modify this file? Is there any automation that will detect a mismatch between supported resources and resources listed in the ClusterServiceVersion file?
2. Having to duplicate descriptions of resources as you pointed out is awkward.
3. Having our README duplicated in here is also awkward (because presumably we'll update it and it'll get out of date).

Reply:
I'll file an issue for it. The good part is that a new resource will still get included in the bundle, just with a blank description in the creation UI - not ideal but not a deal-breaker either. The descriptions aren't actually duplicates... but only because the current doc comments on the types are pretty generic at the moment. Moving them over should be part of that work too.

The README isn't quite a duplicate of our normal one because the install instructions are specific to OLM - that's also not ideal. I'll make an issue for that too.